include/oi/IntrospectionResult-inl.h

```diff
@@ -70,6 +72,18 @@ inline const result::Element* IntrospectionResult::const_iterator::operator->()
   return &*next_;
 }
 
+inline IntrospectionResult::const_iterator
+IntrospectionResult::const_iterator::clone() const {
+  auto ret{*this};
+
+  // Fix the self referential type_path_ field in next_
+  if (ret.next_.has_value()) {
+    ret.next_->type_path = ret.type_path_;
+  }
```

Comment:
I wouldn't mind if this was in the private copy constructor but I don't want to lose the `= default` for it. I don't think there's any way to say use the default then run an extra bit of code directly in the copy constructor - please correct me if I'm wrong.